oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java

```diff
@@ -94,21 +92,35 @@ protected ACTION getUserAction(Cluster cluster, Path buildPath) throws FalconExc
         addOutputFeedsAsArgument(argList, cluster);
         addInputFeedsAsArgument(argList, cluster);
 
-        sparkAction.setJar(addUri(sparkFilePath, cluster));
-
-        setSparkLibFileToWorkflowLib(sparkFilePath, entity);
+        // In Oozie spark action, value for jar is either Java jar file path or Python file path.
+        validateSparkJarFilePath(sparkJarFilePath);
+        sparkAction.setJar(getSparkJarFileName(sparkJarFilePath));
+        setSparkLibFileToWorkflowLib(sparkJarFilePath, entity);
```

Comment:
just pass sparkJarFilePath.toString less number of changes will be required in the code.